internal/runners/activate/namespace.go

```diff
@@ -19,118 +16,78 @@ import (
 	"github.com/ActiveState/cli/internal/locale"
 )
 
-// NamespaceSelect will select the right directory associated with a namespace, and chdir into it
+// NamespaceSelect will select the right directory associated with a namespace
 type NamespaceSelect struct {
-	config   *config.Instance
-	prompter prompt.Prompter
+	config *config.Instance
 }
 
-func NewNamespaceSelect(config *config.Instance, prime primeable) *NamespaceSelect {
-	return &NamespaceSelect{config, prime.Prompt()}
+func NewNamespaceSelect(config *config.Instance) *NamespaceSelect {
+	return &NamespaceSelect{config}
 }
 
-func (r *NamespaceSelect) Run(namespace string, preferredPath string) (string, error) {
+func (r *NamespaceSelect) Run(namespace *project.Namespaced, preferredPath string) (string, error) {
 	// Detect targetPath either by preferredPath or by prompting the user
 	targetPath := preferredPath
 	if targetPath == "" {
 		var err error
-		targetPath, err = r.promptForPath(namespace)
+		targetPath, err = r.getProjectPath(namespace)
 		if err != nil {
 			return "", err
 		}
 	}
 
-	// Validate that target path doesn't contain a config for a different namespace
-	if err := r.validatePath(namespace, targetPath); err != nil {
-		return "", err
-	}
-
 	err := fileutils.MkdirUnlessExists(targetPath)
 	if err != nil {
 		return "", err
 	}
 
-	return targetPath, nil
-}
-
-func (r *NamespaceSelect) promptForPath(namespace string) (string, error) {
-	// If no targetPath was given try to get it from our config (ie. previous activations)
-	paths := projectfile.GetProjectPaths(r.config, namespace)
-	if len(paths) > 0 {
-		targetPath, err := r.promptAvailablePaths(paths)
-		if err != nil {
-			return "", err
-		}
-		if targetPath != nil {
-			return *targetPath, nil
-		}
-	}
-
-	// Is targetPath STILL nil? Prompt the user for a path
-	userPath, err := r.promptForPathInput(namespace)
-	if err != nil {
+	// Validate that target path doesn't contain a config for a different namespace
+	if err := r.validatePath(namespace.Project, targetPath); err != nil {
 		return "", err
 	}
 
-	return userPath, nil
+	return targetPath, nil
 }
 
-func (r *NamespaceSelect) promptAvailablePaths(paths []string) (*string, error) {
-	if len(paths) == 0 {
-		return nil, nil
+func (r *NamespaceSelect) getProjectPath(namespace *project.Namespaced) (string, error) {
+	paths := projectfile.GetProjectPaths(r.config, namespace.String())
+	if len(paths) > 0 {
+		return paths[0], nil
 	}
 
-	noneStr := locale.T("activate_select_optout")
-	choices := append(paths, noneStr)
-	var defaultPath string
-	path, err := r.prompter.Select(locale.Tl("activate_existing_title", "Existing Checkout"), locale.T("activate_namespace_existing"), choices, &defaultPath)
+	targetPath, err := getSafeWorkDir()
 	if err != nil {
-		return nil, err
-	}
-	if path != "" && path != noneStr {
-		return &path, nil
+		return "", locale.NewError("err_get_wd", "Could not get safe working directory")
```

Comment:
```suggestion
		return "", locale.NewError("err_get_wd", "Could not get safe working directory")
```

Please write your localized errors with end-users in mind. This would not be a very helpful error to encounter.